cpp/src/slice2cs/Gen.cpp

```diff
@@ -26,18 +26,83 @@ using namespace IceInternal;
 
 namespace
 {
-    void writeDocLines(Output& out, const StringList& lines)
+    void
+    writeDocLine(Output& out, const string& openTag, const string& comment, std::optional<string> closeTag = nullopt)
     {
-        for (const auto& line : lines)
+        if (comment.empty())
         {
-            out << nl << "///";
-            if (!line.empty())
+            return;
+        }
+
+        if (!closeTag)
+        {
+            closeTag = openTag;
+        }
+
+        out << nl << "/// <" << openTag << ">" << comment << "</" << *closeTag << ">";
+    }
+
+    void
+    writeDocLines(Output& out, const string& openTag, const StringList& lines, optional<string> closeTag = nullopt)
+    {
+        // If there is a single line, write the doc-comment as a single line. Otherwise, write the doc-comment on
+        // multiple lines.
+
+        if (lines.size() == 1)
+        {
+            writeDocLine(out, openTag, lines.front(), closeTag);
+        }
+        else
+        {
+            if (lines.empty())
             {
-                out << " " << line;
+                return;
             }
+
+            if (!closeTag)
+            {
+                closeTag = openTag;
+            }
+
+            bool firstLine = true;
+
+            for (const auto& line : lines)
+            {
+                if (firstLine)
+                {
+                    firstLine = false;
+                    out << nl << "/// <" << openTag << ">";
+                }
+                else
+                {
+                    out << nl << "///";
+                    if (!line.empty())
+                    {
+                        out << ' ';
+                    }
+                }
+                out << line;
+            }
+            out << "</" << *closeTag << ">";
         }
     }
 
+    // Standard marshal doc-comment
```

Comment:
Why not make this a doc-comment, so we could see it on hover?
```suggestion
    /// Standard marshal doc-comment
```